src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -41,9 +41,14 @@
      * login page directly), this returns the URL the user should be redirected
      * to after logging in successfully (e.g. your homepage).
      *
+     * @deprecated Implement onAuthenticationSuccess() instead of needing this function.
+     *
      * @return string
      */
-    abstract protected function getDefaultSuccessRedirectUrl();
+    protected function getDefaultSuccessRedirectUrl()
```

Comment:
Ok, I've removed it :). The problem is purely for *new* users of this class: they technically won't be required at a PHP-level to implement `getDefaultSuccessRedirectUrl` *or* `onAuthenticationSuccess`. I was just trying to avoid an ugly "method not found" call. I've actually still handled this, with an `if` statement check.